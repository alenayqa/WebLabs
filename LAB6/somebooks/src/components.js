import {sum, mult, inc} from './functions.js'

export const sumDiv = ({props}) =>
{
    return <div>
    <b>{props.a}</b>
    <b>{props.b}</b>
    <b>{sum(props.a, props.b)}</b>
    </div>
}

export const multDiv = ({props}) =>
{
    return <div>
    <b>{props.a}</b>
    <b>{props.b}</b>
    <b>{mult(props.a, props.b)}</b>
    </div>
} 

export const incDiv = ({props}) =>
{
    return <div>
    <b>{props.a}</b>
    <b>{inc(props.a)}</b>
    </div>
} 