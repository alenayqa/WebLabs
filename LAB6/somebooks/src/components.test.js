import { sumDiv, multDiv, incDiv } from "./components";
import renderer from 'react-test-renderer';

test('sum div render test', ()=>{
    const result = renderer.create(<sumDiv
        a = {5}
        b = {2}
        />);
        expect(result).toMatchSnapshot();
})

test('mult div render test', ()=>{
    const result = renderer.create(<multDiv
        a = {5}
        b = {2}
        />);
        expect(result).toMatchSnapshot();
})

test('inc div render test', ()=>{
    const result = renderer.create(<incDiv
        a = {5}
        />);
        expect(result).toMatchSnapshot();
})