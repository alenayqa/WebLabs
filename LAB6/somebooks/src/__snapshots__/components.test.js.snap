// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inc div render test 1`] = `
<incDiv
  a={5}
/>
`;

exports[`mult div render test 1`] = `
<multDiv
  a={5}
  b={2}
/>
`;

exports[`sum div render test 1`] = `
<sumDiv
  a={5}
  b={2}
/>
`;
